File: _archive_tests/analyze_sheets.ts
import * as XLSX from 'xlsx';
import * as fs from 'fs';
import * as path from 'path';

const directory = "/home/matheus/Área de trabalho/ERP-Geral/revestimentos/planilhas";
const files = [
    "BOUTIQUE BRASIL 2025 - VAREJO PIERINI.xlsx",
    "Tabela Preços Castelli - Tab01 - +70m2.xlsx",
    "Tabela Preço Embramaco Porcelanato - Tab01_Nova Politica_Jan2025.xls"
];

files.forEach(filename => {
    console.log(`\n\n--- Analyzing: ${filename} ---`);
    const filePath = path.join(directory, filename);

    try {
        const workbook = XLSX.readFile(filePath);
        const sheetName = workbook.SheetNames[0];
        const sheet = workbook.Sheets[sheetName];

        // Convert to array of arrays
        const data = XLSX.utils.sheet_to_json(sheet, { header: 1, range: 0, defval: "" });

        console.log(`First 5 rows (to identify header):`);
        for (let i = 0; i < Math.min(10, data.length); i++) {
            console.log(`Row ${i}:`, JSON.stringify(data[i]));
        }

    } catch (e) {
        console.error(`Error processing ${filename}:`, e.message);
    }
});

File: _archive_tests/analyze_sheets.py
import pandas as pd
import os

directory = "/home/matheus/Área de trabalho/ERP-Geral/revestimentos/planilhas"
files = [
    "BOUTIQUE BRASIL 2025 - VAREJO PIERINI.xlsx",
    "Tabela Preços Castelli - Tab01 - +70m2.xlsx",
    "Tabela Preço Embramaco Porcelanato - Tab01_Nova Politica_Jan2025.xls"
]

for filename in files:
    path = os.path.join(directory, filename)
    print(f"\n--- Analyzing: {filename} ---")
    try:
        # Try finding header row automatically or default to 0
        # Often these sheets have metadata in top rows. I'll read first 10 rows and print them to visually identify header.
        df_preview = pd.read_excel(path, header=None, nrows=10)
        print("First 10 rows preview:")
        print(df_preview.to_string())
    except Exception as e:
        print(f"Error reading {filename}: {e}")
